archive/app/controllers/ArchiveController.scala

```diff
@@ -109,34 +108,40 @@ class ArchiveController(redirects: RedirectService) extends Controller with Logg
   }
 
   private object Century {
-    private val CenturyUrlEx = """www.theguardian.com\/century(\/)?$""".r
-    private val CenturyDecadeUrlEx = """www.theguardian.com(\/\d{4}-\d{4})(\/)?$""".r
-    private val CenturyStoryUrlEx = """www.theguardian.com\/(\d{4}-\d{4})\/Story\/([0|1]?,[\d]*,-?\d+,[\d]*)(.*)""".r
-    private val ngCenturyFront = "www.theguardian.com/world/2014/jul/31/-sp-how-the-guardian-covered-the-20th-century"
+    private val CenturyUrlEx = """/century(\/)?$""".r
+    private val CenturyDecadeUrlEx = """(\/\d{4}-\d{4})(\/)?$""".r
+    private val CenturyStoryUrlEx = """\/(\d{4}-\d{4})\/Story\/([0|1]?,[\d]*,-?\d+,[\d]*)(.*)""".r
+    private val ngCenturyFront = "/world/2014/jul/31/-sp-how-the-guardian-covered-the-20th-century"
 
     def unapply(path: String): Option[String] = path match {
       case CenturyUrlEx(_) => Some(ngCenturyFront)
       case CenturyDecadeUrlEx(_, _) => Some(ngCenturyFront)
-      case CenturyStoryUrlEx(decade, storyId, ext) => Some(s"www.theguardian.com/century/$decade/Story/$storyId$ext")
+      case CenturyStoryUrlEx(decade, storyId, ext) => Some(s"/century/$decade/Story/$storyId$ext")
       case _ =>  None
     }
   }
 
   private object Lowercase {
     def unapply(path: String): Option[String] = path.split("/").toList match {
-        case "www.theguardian.com" :: section :: other if section.exists(_.isUpper) =>
-          Some(s"www.theguardian.com/${section.toLowerCase}/${other.mkString("/")}")
+        case "" :: section :: other if section.exists(_.isUpper) =>
+          Some(s"/${section.toLowerCase}/${other.mkString("/")}")
         case _ => None
     }
   }
 
-  private def redirectTo(path: String, identifier: String)(implicit request: RequestHeader): Result = {
-    log.info(s"""Archive $redirectHttpStatus, "$identifier" redirect to $path""")
-    Cached(CacheTime.ArchiveRedirect)(WithoutRevalidationResult(Redirect(s"http://$path", redirectHttpStatus)))
+  private def redirectTo(path: String, identifier: String, pathSuffixes: String*)(implicit request: RequestHeader): Result = {
+    val endOfPath = if(pathSuffixes.isEmpty) "" else s"/${pathSuffixes.mkString("/")}"
+    val redirect = LinkTo(path) + endOfPath
+
+    log.info(s"""Archive $redirectHttpStatus, "$identifier" redirect to $redirect""")
+    Cached(CacheTime.ArchiveRedirect)(WithoutRevalidationResult(Redirect(redirect, redirectHttpStatus)))
```

Comment:
I can see that it would help to quickly diagnose which of the custom redirects in the controller was used.  Whether we ever really need that info - I'm not so sure.  It may have been useful when first adding these redirects, but I would be ok with removing now.